### PR TITLE
Adds `from_structure` staticmethods

### PR DESCRIPTION
**Pull request by colinschoen**
Allows `Imposter`s and related types to be created from the output
of `as_structure`.

`Proxy` modified to be used as a Response inside of a Stub. This
is more consistent with the mountebank contract.

### CODE REVIEW

**Comment by colinschoen** [comment]
CC: @sgupta @jnb

File: tests/integration/test_proxy.py [hunk 1]
```diff
@@ -8,14 +8,14 @@
...
-from mbtest.imposters import Imposter, Proxy
+from mbtest.imposters import Imposter, Proxy, Stub
 from mbtest.matchers import had_request
 
 logger = logging.getLogger(__name__)
 
 
 def test_proxy(mock_server):
-    imposter = Imposter(Proxy(to="http://example.com"))
+    imposter = Imposter(Stub(responses=Proxy(to="http://example.com")))
```
Comment (brunns) [inline tests/integration/test_proxy.py]: Can we do this change in a backwards compatible way?

File: src/mbtest/imposters.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ class Protocol(Enum):
         HTTPS = "https"
         SMTP = "smtp"
 
-    def __init__(self, stubs, port=None, protocol=Protocol.HTTP, name=None, record_requests=True):
+    def __init__(self, stubs, port=4545, protocol=Protocol.HTTP, name=None, record_requests=True):
```
[the remark this answers is not in the document]
Comment (colinschoen) [reply src/mbtest/imposters.py]: Whoops, I thought I reverted this change. I've done so now.

File: tests/integration/test_proxy.py (hunk 1 above)
Comment (colinschoen) [reply tests/integration/test_proxy.py]: Good point. I've implemented chat.

**Comment by brunns** [comment]
Merged. I've moved the tests around a bit, too.

**Pull request merged**